File: sis-kitnet/src/app/core/model/dto/controleLancamento/controleLancamentoResponseDTO.ts
import { StatusDTO } from './statusDTO';
import { ValorRegraDTO } from './valorRegraDTO';
import { ValorId } from './valorId';
import { InquilinoId } from './inquilinoId';
import { ApartamentoId } from './apartamentoId';

export interface ControleLancamentoResponseDTO {
  id: number;
  dataLancamento: string;
  dataEntrada: string;
  dataPagamento?: string;
  observacao?: string;
  status?: StatusDTO;
  valores?: ValorRegraDTO;
  valor: ValorId;
  inquilino: InquilinoId;
  apartamento: ApartamentoId;
}

File: kitnet/src/main/java/kitnet/com/api/dto/controleLancamento/ControleLancamentoResponseDTO.java
package kitnet.com.api.dto.controleLancamento;

import kitnet.com.api.dto.apartamento.ApartamentoId;
import kitnet.com.api.dto.inquilino.InquilinoId;
import kitnet.com.api.dto.valor.ValorId;

import java.time.LocalDate;

public record ControleLancamentoResponseDTO(
    Long id,
    LocalDate dataLancamento,
    LocalDate dataEntrada,
    LocalDate dataPagamento,
    String observacao,
    StatusDTO status,
    ValorRegraDTO valores,
    ValorId valor,
    InquilinoId inquilino,
    ApartamentoId apartamento,
    String nomeInquilino,
    String numeroApartamento
) {}
